### ISSUE TITLE
Issue with rendering math on multiple lines

### ISSUE THREAD

**Issue by tlyon3** [nteract/archived-desktop-app#2924]
The renderer runs into issues with math that it split on multiple lines with the `\` character:
``` 
$$ \frac{d^2\vec{r_0}}{dt^2} = G \, m_1 \frac{\vec{r_1} - \vec{r_0}}{{\vert \vec{r_1} - \vec{r_0} \vert}^3} + \
                               G \, m_2 \frac{\vec{r_2} - \vec{r_0}}{{\vert \vec{r_2} - \vec{r_0} \vert}^3} $$
```
The first line of the math is removed and only the second is displayed. I think it's because in the actual json for the notebook, this block of math is represented in two separate strings:
```
"$$ \\frac{d^2\\vec{r_0}}{dt^2} = G \\, m_1 \\frac{\\vec{r_1} - \\vec{r_0}}{{\\vert \\vec{r_1} - \\vec{r_0} \\vert}^3} + \\\n",
```
and

```
"                               G \\, m_2 \\frac{\\vec{r_2} - \\vec{r_0}}{{\\vert \\vec{r_2} - \\vec{r_0} \\vert}^3} $$\n",
```

Because the first string doesn't have a closing `$$` (and the second an opening `$$`) the parser could be throwing it away as invalid syntax. This is likely a bug with the `remark-math` plugin with it only parsing string by string and not accounting for math that could be separated in two separate strings.

**Comment by ghost** [comment]
This code may be considered malformed, you could use this pandoc [rule](http://pandoc.org/MANUAL.html#extension-tex_math_dollars) from Extension: `tex_math_dollars` to parse rigorously maths:

> Anything between two $ characters will be treated as TeX math. The opening $ must have a non-space character immediately to its right, while the closing $ must have a non-space character immediately to its left, and must not be followed immediately by a digit. Thus, $20,000 and $30,000 won’t parse as maths. If for some reason you need to enclose text in literal $ characters, backslash-escape them and they won’t be treated as maths delimiters.

Or if we consider mathjax is right as it parses this code correctly implement a parser like [tex2jax.js](https://github.com/mathjax/MathJax/blob/master/unpacked/extensions/tex2jax.js) used in mathjax.

**Comment by x1angli** [comment]
Viewing malformed math formulas indeed bothers me (and others, I believe)

A very common case: 
Normally MathJax does NOT require `$$` signs enclosing math expressions to lead or follow line-breaks. That is to say, if a math expression wrapped by a pair of in-line starting and closing $$ signs. 

But, the particular Mathjax shipped in nteract does have such peculiar requirements, therefore nteract cannot render these math expressions properly (see the pictures) 

![Capture](https://user-images.githubusercontent.com/8603485/56088821-dad79400-5eba-11e9-8cd0-0180611eaa47.PNG)
![Capture1](https://user-images.githubusercontent.com/8603485/56088822-dad79400-5eba-11e9-8b6a-976a98825ee7.PNG)


Unfortunately, there are tons of such math expressions out there, which is really bothersome for nteract users like me.....


Fixing this would be really simple: just upgrade the MathJax shipped in nteract to the latest one.

Could anybody here to help? Thanks a lot

**Comment by BenRussert** [comment]
@x1angli could you paste that snippet into a comment?

**Comment by x1angli** [comment]
> @x1angli could you paste that snippet into a comment?

you can use any math expression to reproduce this issue, as long as the beginning double dollar sign `$$` and the ending one is not put in a separate line. 

something like 
```
$$ 1 + 1 = 2 $$ some text comments in the same line
```

~~All I'm proposing is just bumping the MathJax's version~~

**Comment by BenRussert** [comment]
> All I'm proposing is just bumping the MathJax's version.

@x1angli would you want to send a PR to fix?

@lgeiger, wonder if you have thoughts on this?

**Comment by x1angli** [comment]
> > All I'm proposing is just bumping the MathJax's version.
> 
> @x1angli would you want to send a PR to fix?

Sorry, please disregard that part. 

After revisiting this issue, I found it more complicated than bumping a version number.

Rephrase the malformed issue

```
some text 
$$
a + b = c $$    the follow line should be rendered as plain-text, rather than Latex math
```

Actually, nteract-on-jupyter (the web one) has a condensed/minified .js file with a version number of 2.3.5, which is the latest.  I am unable to check the version number of the Electron packed desktop one ,  but given  that the malformed rendering is reproducible on both web and desktop, I assume they both use 2.3.5.

However, it is hard to find this issue on any other editors that use Mathjax --- those editors work very well.

So I guess this might be caused by the interaction/incompatibility between Mathjax and another js file loaded on a nteract page. Something messy kicked in and scrambled Mathjax's parsing code.

Please excuse me that the analysis above is all I can offer. I tried my best to provide more insights, but I'm not familiar w/ either JS rendering or node.js.  Hope it's helpful. thanks

**Comment by captainsafia** [comment]
Keeping this closed but adding a ref to #5023.
